File: LeetCode/Medium/3517-smallest-palindromic-rearrangement-i/3517-smallest-palindromic-rearrangement-i.java
class Solution {
    public String smallestPalindrome(String s) {
        int[] freq = new int[26];
        for (char ch : s.toCharArray()) {
            freq[ch - 'a']++;
        }        
        String l = "";
        String m = "";
        for (int i = 0; i < 26; i++) {
            if (freq[i] == 0) continue;
            char ch = (char) (i + 'a');
            if (freq[i] % 2 != 0) {
                m = String.valueOf(ch);
            }
            l += String.valueOf(ch).repeat(freq[i] / 2);
        }
        String r = new StringBuilder(l).reverse().toString();
        return l + m + r;
    }
}

File: LeetCode/Medium/3517-smallest-palindromic-rearrangement-i/3517-smallest-palindromic-rearrangement-i.cpp
class Solution {
public:
    string smallestPalindrome(string s) {
        int freq[26] = {0};
        for (char ch : s) 
            freq[ch - 'a']++;
        string l = "";
        string m = "";
        for(int i = 0; i < 26; i++){
            if(freq[i] == 0) continue;
            char ch = i + 'a';
            if(freq[i] % 2 != 0) {
                m = ch;
            }
            l.append(freq[i] / 2, ch);
        }
        string r = l;
        reverse(r.begin(), r.end());
        return l + m + r;
    }
};
